### PR TITLE
Implement support for no_std

### PR DESCRIPTION
**Pull request by CryZe**
It's a pretty simple fix.

### CODE REVIEW

**Comment by CryZe** [comment]
When testing there's some weird problem where it says it can't find the global allocator. It doesn't cause the tests to fail, but it still shows it. No idea how to get rid of that.

File: src/lib.rs [hunk 1]
```diff
@@ -1,6 +1,12 @@
+#![no_std]
```
Comment (RReverser) [inline src/lib.rs]: Will this work with `std` as well? I thought in the past crates needed to have a separate Cargo feature for this mode?

Comment (CryZe) [reply src/lib.rs]: Yes it works with std as well.

Comment (CryZe) [reply src/lib.rs]: You only need an std feature for enabling std-only functionality.

File: src/lib.rs [hunk 2]
```diff
@@ -24,7 +30,7 @@ pub trait Pattern<'s> {
 macro_rules! impl_pattern {
 	($ty:ty $(where $($bound:tt)*)?) => {
 		impl<'s $(, $($bound)*)?> Pattern<'s> for $ty {
-			type MatchIndices = std::str::MatchIndices<'s, Self>;
+			type MatchIndices = core::str::MatchIndices<'s, Self>;
```
Comment (RReverser) [inline src/lib.rs]: Do references in doc comments need to be updated as well? (I don't know how / whether anyone builds docs with `no_std`)

Comment (CryZe) [reply src/lib.rs]: Considering core's docs also just refer to std, I didn't bother changing them. Idk, it's a bit weird.

**Comment by RReverser** [comment]
> When testing there's some weird problem where it says it can't find the global allocator. It doesn't cause the tests to fail, but it still shows it. No idea how to get rid of that.

I guess for now you could use something like `#![cfg_attr(not(doctest), no_std)]`?

**Comment by RReverser** [comment]
So yeah, the error is essentially #52243 and seems fixed on nightly. For now, let's just go with an attribute like above.

File: src/lib.rs (hunk 2 above)
Comment (RReverser) [reply src/lib.rs]: I've changed the extern crate reference to `extern crate alloc as std` - it seems less confusing and more consistent that way and allows to keep all `std::` references as-is.

File: src/lib.rs (hunk 1 above)
Comment (RReverser) [reply src/lib.rs]: Thanks.

**Pull request merged**